src/components/Utils/utils.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import FilterContainer from '@ferlab/ui/core/components/filters/FilterContainer';
+import FilterSelector from '@ferlab/ui/core/components/filters/FilterSelector';
+import { IFilter, IFilterGroup } from '@ferlab/ui/core/components/filters/types';
+import {
+  getFilterType,
+  getSelectedFilters,
+  updateFilters,
+} from '@ferlab/ui/core/data/filters/utils';
+
+import history from 'services/history';
+import { keyEnhance, keyEnhanceBooleanOnly, underscoreToDot } from 'store/graphql/utils';
+import { ExtendedMapping, MappingResults } from 'store/graphql/utils/actions';
+
+export type Results = {
+  data: GQLData | null;
+  loading: boolean;
+};
+
+export interface RangeAggs {
+  stats: {
+    max: number;
+    min: number;
+  };
+}
+export interface TermAggs {
+  buckets: TermAgg[];
+}
+
+export interface TermAgg {
+  doc_count: number;
+  key: string;
+}
+
+export type Aggs = TermAggs | RangeAggs;
+
+const isTermAgg = (obj: any): obj is TermAggs => obj.buckets !== undefined;
+const isRangeAgg = (obj: any): obj is RangeAggs => obj.stats !== undefined;
+
+export interface GQLData<T extends Aggs = any> {
+  hits: any; //TODO refine type?
```

Comment:
TODO have been done ?

Reply:
No I've removed the TODO. I think it would be a huge variant type. that would have to be changed if the data changes.